src/main/java/org/codehaus/mojo/license/model/LicenseMap.java

```diff
@@ -158,19 +158,31 @@ public LicenseMap toLicenseMapOrderByName()
         return result;
     }
 
-    public void removeProject( MavenProject project )
+    /**
+     * Remove project from all licenses it is mapped to.
+     * @param project
+     * @return a List of license names that the given project was mapped to
+     */
+    public List<String> removeProject( MavenProject project )
     {
+        List<String> removedFrom = new ArrayList<>();
         for ( Map.Entry<String, SortedSet<MavenProject>> entry : entrySet() )
         {
             SortedSet<MavenProject> projects = entry.getValue();
             for ( MavenProject mavenProject : projects )
             {
                 if ( project.equals( mavenProject ) )
                 {
-                    get( entry.getKey() ).remove( project );
-                    return;
+                    removedFrom.add( entry.getKey() );
                 }
             }
         }
+
+        for ( String r : removedFrom )
+        {
+            get( r ).remove( project );
+        }
+
+        return removedFrom;
```

Comment:
Removing the entries directly yields a ConcurrentModificationException. Also, I collect the removedFrom licenses for logging.